### DISCUSSION TITLE
Wrong result verify password

### DISCUSSION THREAD

**Discussion by letanloc1998** [fastapi/fastapi#6943] [Questions] [answered]
### First check

* [x] I added a very descriptive title to this issue.
* [x] I used the GitHub search to find a similar issue and didn't find it.
* [x] I searched the FastAPI documentation, with the integrated search.
* [x] I already searched in Google "How to X in FastAPI" and didn't find any information.
* [x] I already read and followed all the tutorial in the docs and didn't find an answer.
* [ ] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
* [ ] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
* [ ] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).
* [ ] After submitting this, I commit to one of:
    * Read open issues with questions until I find 2 issues where I can help someone and add a comment to help there.
    * I already hit the "watch" button in this repository to receive notifications and I commit to help at least 2 people that ask questions in the future.
    * Implement a Pull Request for a confirmed bug.

```python
from passlib.context import CryptContext

pwd_context = CryptContext(schemes=["bcrypt"], deprecated="auto")


def verify_password(plain_password, hashed_password):
    return pwd_context.verify(plain_password, hashed_password)


def get_password_hash(password):
    return pwd_context.hash(password)
```
### Example

Here's a self-contained, [minimal, reproducible, example](https://stackoverflow.com/help/minimal-reproducible-example) with my use case:

```python
token = "2579353e88cf33776499aeda887226fb5e6999bdff129e45a24943f8ea48c784f6faf5d60942162a7d2395b4742037243fa41ffd5f6f3a23672e8c1f76fdf2723ed799101d"
verify_password(token + "b", get_password_hash(token + "a"))
```

***Expected result: False***
***Actual result: True***

![image](https://user-images.githubusercontent.com/26348718/103496318-5556fd00-4e70-11eb-82cb-59facd9be736.png)

### Additional context

https://fastapi.tiangolo.com/tutorial/security/oauth2-jwt/#hash-and-verify-the-passwords

**Comment by Mause** [comment, marked as answer]
> Many implementations of bcrypt truncate the password to the first 72 bytes, following the OpenBSD implementation.

https://en.wikipedia.org/wiki/Bcrypt

**Comment by letanloc1998** [comment]
> > Many implementations of bcrypt truncate the password to the first 72 bytes, following the OpenBSD implementation.
> 
> https://en.wikipedia.org/wiki/Bcrypt

Thank you

**Comment by tiangolo** [comment]
Thanks for the help here @Mause ! 👏  🙇 

Thanks for reporting back and closing the issue @letanloc1998  👍

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.
